js/src/views/insert-shortcode-list.js

```diff
@@ -6,37 +7,84 @@ var insertShortcodeListItem = require('sui-views/insert-shortcode-list-item');
 var insertShortcodeList = wp.Backbone.View.extend({
 
 	tagName : 'div',
+	className : 'insert-shortcode-list',
 	template : wp.template('add-shortcode-list'),
 
+	events: {
+		'click .shortcode-list-item': 'selectShortcode',
+	},
+
 	initialize : function( options ) {
+		this.setShortcodes( ( 'shortcodes' in options ) ? options.shortcodes : [] );
+		this.refresh();
+	},
 
-		this.displayShortcodes( options );
+	/**
+	 * Set / Update shortcodes list.
+	 */
+	setShortcodes: function( shortcodes ) {
 
-	},
-	
-	refresh: function( shortcodeData ) {
-		var options;
-		if ( shortcodeData instanceof Backbone.Collection ) {
-			options = { shortcodes: shortcodeData };
+		if ( shortcodes instanceof Shortcodes ) {
+			this.shortcodes = shortcodes;
+		} else if ( Array.isArray( shortcodes ) ) {
+			this.shortcodes = new Shortcodes( shortcodes );
 		} else {
-			options = { shortcodes: new Shortcodes( shortcodeData ) };
+			this.shortcodes = new Shortcodes();
 		}
-		this.displayShortcodes( options );
+
 	},
-	
-	displayShortcodes: function(options) {
-		var t = this;
-		
-		t.$el.find('.add-shortcode-list').html('');
-		t.options = {};
-		t.options.shortcodes = options.shortcodes;
-
-		t.options.shortcodes.each(function(shortcode) {
-			t.views.add('ul', new insertShortcodeListItem({
+
+	selectShortcode: function(e) {
+
+		var target    = $( e.currentTarget );
+		var shortcode = this.shortcodes.findWhere( { shortcode_tag: target.attr( 'data-shortcode' ) } );
+
+		if ( shortcode ) {
+			this.trigger( 'shortcode-ui:select', shortcode );
+		}
+
+	},
+
+	/**
+	 * Refresh & render shortcodes and sub-views.
+	 */
+	refresh: function( shortcodes ) {
+
+		shortcodes = shortcodes || this.shortcodes;
+
+		// Remove existing views.
+		_.each( this.views.get('ul'), function( view ) {
+			view.remove();
+		} );
+
+		shortcodes.each( function( shortcode ) {
+			this.views.add( 'ul', new insertShortcodeListItem({
 				model : shortcode
 			}));
-		});
-	}
+		}.bind(this) );
+
+	},
+
+	search: function( s ) {
+
+		if ( s && s.length ) {
+
+			var pattern = s.replace( /[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&" );
```

Comment:
Problem is that currently we just pass the whole search query when creating to the regex. Try seaching for `\` or `*` and you get a JS error. 

This is trying to escape anything in the search query before using to create the Regex.